### PR TITLE
nickreq typescript rewrite

### PR DESCRIPTION
**Pull request by Aman25ta**
@DankDumpster

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+	"token": "",
+	"botId": "",
+	"channelid": "",
+	"color": "0x20d84e",
+	"ownerid": "",
+	"postgres": {
+		"username": "",
+		"password": "",
+		"host": "",
```
Comment (AltF02) [inline config.json]: Could you possibly use .yml files for config?

File: package.json [hunk 2]
```diff
@@ -1,25 +1,14 @@
 {
-  "scripts": {
-    "start": "tsc && node build/."
-  },
   "dependencies": {
-    "discord.js": "^12.5.1",
-    "discord.js-commando": "github:discordjs/commando",
-    "dotenv": "^8.2.0"
+    "@newcircuit/logging.js": "git+https://github.com/NewCircuit/logging.js.git",
```
Comment (AltF02) [inline package.json]: You can remove this for now

File: src/commands/nickreq/nickreq.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+			description: 'Main Nickname Request commands',
+			argstype: 'single'
+		})
+	}
+	async run(message, args) {
+		if (message.guild == null) return
+		if (args.length > 32) {
+			return await message.reply("The nickname must be less than 32 characters");
+		}
+		if (args.length < 1) {
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: Its possible to provide args in the constructor, example:

```ts
      args: [
        {
          key: 'nickname',
          prompt: 'What is the nickname you want to request?',
          type: 'string',
          infinite: true,
        },
      ], 
```

More info here: https://discordjs.guide/commando/args.html#string-arguments

File: src/commands/nickreq/nickreq.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		    .setStyle("green")
+		    .setID("Accepted");
+		let button2 = new MessageButton()
+				.setLabel("Reject")
+		    .setStyle("red")
+		    .setID("Rejected");
+		let row = new MessageActionRow()
+		  .addComponent(button1)
+		  .addComponent(button2);
+		await channel.send({
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: Nit: This indentation is inconsistent

File: src/db.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+// pool.connect()
+// .then(() => console.log("Connected"))
+// .then(() => pool.query("DROP TABLE nickreq"))
+// .then(() => pool.query(`CREATE TABLE nickreq (
+// 	user_id VARCHAR ( 50 ) NOT NULL,
+// 	nick VARCHAR ( 50 ) NOT NULL,
+// 	state BOOLEAN
+// );`))
+// .catch(err => console.error(err))
+// .finally(() => pool.end())
```
Comment (AltF02) [inline src/db.ts]: Could you possibly make an sql file? For example in

migrations/init.sql

File: src/db.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+async function reject(userid) {
+	userid = userid.toString()
+	const results = await pool.query("UPDATE nickreq SET state=$1 WHERE user_id = $2 AND state IS NULL",[0 , userid]);
+	return results.rows
+}
+
+module.exports.check = check;
+module.exports.insert = insert;
+module.exports.accept = accept;
+module.exports.reject = reject;
```
Comment (AltF02) [inline src/db.ts]: Nit: No EOF

File: tsconfig.json [hunk 7]
```diff
@@ -1,70 +1,16 @@
...
-    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
-
-    /* Experimental Options */
-    // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
-    // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
-
-    /* Advanced Options */
-    "skipLibCheck": true,                     /* Skip type checking of declaration files. */
-    "forceConsistentCasingInFileNames": true  /* Disallow inconsistently-cased references to the same file. */
+    "target": "es6",
```
Comment (AltF02) [inline tsconfig.json]: Note: You can target es2020, if you require certain features

File: src/index.ts [hunk 8]
```diff
@@ -1,27 +1,64 @@
...
-  ])
-  .registerDefaults()
-  .registerCommandsIn(path.join(__dirname, 'commands'));
 
-client.login(process.env.DISCORD_TOKEN);
+
+
+client.registry.registerGroup('nickreq','nickreq commands')
+.registerDefaults()
+.registerCommandsIn(path.join(__dirname,"commands"));
```
Comment (AltF02) [inline src/index.ts]: Nit: Indentation

File: src/db.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+// .then(() => pool.query("DROP TABLE nickreq"))
+// .then(() => pool.query(`CREATE TABLE nickreq (
+// 	user_id VARCHAR ( 50 ) NOT NULL,
+// 	nick VARCHAR ( 50 ) NOT NULL,
+// 	state BOOLEAN
+// );`))
+// .catch(err => console.error(err))
+// .finally(() => pool.end())
+
+async function check(userid) {
```
Comment (AltF02) [inline src/db.ts]: Possibly better to use static methods here?

File: src/commands/nickreq/nickreq.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
+const commando = require("discord.js-commando");
+const discord = require("discord.js");
+const { MessageButton, MessageActionRow } = require('discord-buttons');
+let config = require('../../../config.json');
+let db = require("../../db.js");
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: Could you possibly use import syntax?

File: src/commands/nickreq/nickreq.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+module.exports = class nickReq extends commando.Command {
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: You can do 

```ts
export default class nickReq extends commando.Command
```

**Review by AltF02 — CHANGES_REQUESTED** [review]
Forgot to mention, I recommend using eslint with the airbnb preset. 

Certain rules like `no-console` can be turned off ofc

**Comment by Aman25ta** [comment]
Will do

As for the string args, as im taking everything after, what would be the
advantage of using it, if any?

**Comment by AltF02** [comment]
Yes there would be, as its handled by discord.js commando. So it shows up in the help, also that is accomplished by the 
`infinite: true`

**Comment by Aman25ta** [comment]
@DankDumpster , updated

File: migrations/init.sql [hunk 12]
```diff
@@ -0,0 +1 @@
+OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "mysqldump": executable file not found in $PATH: unknown
```
Comment (AltF02) [inline migrations/init.sql]: Looks like something went wrong here?

File: src/commands/nickreq/nickreq.ts [hunk 13]
```diff
@@ -0,0 +1,78 @@
+const commando = require('discord.js-commando');
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts:1]: Sorry, I mean that preferably you can use `import` syntax and only require if you have no other choice (e.g. the discord-buttons library)

File: src/commands/nickreq/nickreq.ts [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+        name: `${message.author.username}#${message.author.discriminator}`,
+        iconURL: message.author.avatarURL(),
+      },
+      footer: { text: message.author.id },
+    });
+    const button1 = new MessageButton()
+      .setLabel('Accept')
+      .setStyle('green')
+      .setID('Accepted');
+    const button2 = new MessageButton()
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts:66]: Nit: Spacing

File: src/db.ts (hunk 5 above)
Comment (AltF02) [reply src/db.ts]: Looks like this wasn't updated correctly, could you take another look at it?

File: src/index.ts [hunk 15]
```diff
@@ -1,27 +1,72 @@
...
+      title: 'Nickname request',
+      description: button.message.embeds[0].description,
+      fields: [
+        { name: button.id, value: `By ${button.clicker.user.username}#${button.clicker.user.discriminator} (<@${button.clicker.user.id}>)` },
+      ],
+      color: 0x20d84e,
+      author: { name: button.message.embeds[0].author.name, iconURL: button.message.embeds[0].author.iconURL },
+      footer: button.message.embeds[0].footer,
+    },
+  });
```
Comment (AltF02) [inline src/index.ts:48]: Nit: Spacing

**Review by AltF02 — CHANGES_REQUESTED** [review]

**Comment by AltF02** [comment]
@Aman25ta some small nits

File: migrations/init.sql (hunk 12 above)
Comment (Aman25ta) [reply migrations/init.sql]: Weird, ill make another one

File: src/db.ts [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+// .then(() => pool.query("DROP TABLE nickreq"))
+// .then(() => pool.query(`CREATE TABLE nickreq (
+//  user_id VARCHAR ( 50 ) NOT NULL,
+//  nick VARCHAR ( 50 ) NOT NULL,
+//  state BOOLEAN
+// );`))
+// .catch(err => console.error(err))
+// .finally(() => pool.end())
+
+export default class Db {
```
Comment (AltF02) [inline src/db.ts]: NIT: Change to `DB`

File: package.json [hunk 17]
```diff
@@ -1,25 +1,18 @@
...
-    "@types/node": "^14.14.24",
-    "@typescript-eslint/eslint-plugin": "^4.14.2",
-    "@typescript-eslint/parser": "^4.14.2",
-    "eslint": "^7.19.0",
-    "eslint-config-airbnb-base": "^14.2.1",
-    "eslint-plugin-import": "^2.22.1",
-    "typescript": "^4.1.3"
-  },
-  "author": "",
-  "license": "ISC",
```
Comment (AltF02) [inline package.json:23]: Could you add a GPLV3 license?

**Review by AltF02 — CHANGES_REQUESTED** [review]
Looks like something went wrong with your migration again, everything looks good except that. Take a look at https://github.com/NewCircuit/suggestions/blob/master/pg.sql as example

**Comment by Aman25ta** [comment]
@DankDumpster fixed the sql

File: package.json [hunk 18]
```diff
@@ -14,5 +14,6 @@
     "@typescript-eslint/parser": "^4.28.2",
     "eslint": "^7.30.0",
     "eslint-config-airbnb-typescript": "^12.3.1"
-  }
+  },
+  "license": "GPLV3"
```
Comment (AltF02) [inline package.json]: Oh didn't mean it literally, create a LICENSE file more info here:

https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/licensing-a-repository

File: src/db.ts (hunk 5 above)
Comment (AltF02) [reply src/db.ts]: This can be removed now :smile:

**Review by AltF02 — CHANGES_REQUESTED** [review]
Almost there, looks like you forgot to do `git add` on the new sql file tho

**Comment by Aman25ta** [comment]
> Almost there, looks like you forgot to do `git add` on the new sql file tho

i can see the new file in the repo though :thinking:

**Comment by AltF02** [comment]
Yeah you're right, was looking at the wrong thing :sweat_smile:

**Review by AltF02 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**